org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/JavaVariable.java

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+* Copyright (c) 2017 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.adapter;
+
+import com.sun.jdi.Field;
+import com.sun.jdi.LocalVariable;
+import com.sun.jdi.Value;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * This class represents a variable on a stopped stack frame, it contains more informations
+ * about this variable:
+ * <ul>
+ * <li>
+ * The field if this variable is a field value.
+ * </li>
+ * <li>
+ * The array index if this variable is one element of an array.
+ * </li>
+ * <li>
+ * The local variable information if this variable is a local variable.
+ * </li>
+ * <li>
+ * The argument index if this variable is an argument variable.
+ * </li>
+ * </ul>
+ * The about informations are for further formatter to compose an more detailed
+ * name for name conflict situation.
+ */
+public class JavaVariable {
+    /**
+     * The JDI value.
+     */
+    public Value value;
```

Comment:
Let's be really careful of using public fields. I understand these fields act like struct members. Is there a better way like syntactic sugar of handling this situation in Java?